internal/appinfo/appinfo.go

```diff
@@ -22,6 +22,10 @@ type AppInfo struct {
 func execDir(baseDir ...string) (resultPath string) {
 	defer func() {
 		// Account for legacy use-case that wasn't using the correct bin dir
```

Comment:
One of the purposes of this story is that we can get rid of this legacy handling logic. So I would expect to see this defer gone as part of this PR, rather than added to.